core/src/main/java/feign/CollectionFormat.java

```diff
@@ -0,0 +1,20 @@
+package feign;
+
+/**
+ * Various ways to encode collections in URL parameters.
+ *
+ * These specific cases are inspired by the
+ * <a href="http://swagger.io/specification/">OpenAPI specification</a>.
+ */
+public enum CollectionFormat {
+  /** Comma separated values, eg foo=bar,baz */
+  CSV,
+  /** Space separated values, eg foo=bar baz */
+  SSV,
+  /** Tab separated values, eg foo=bar[tab]baz */
+  TSV,
+  /** Values separated with the pipe (|) character, eg foo=bar|baz */
+  PIPES,
+  /** Parameter name repeated for each value, eg foo=bar&foo=baz */
+  MULTI,
```

Comment:
I would either add a package-local method in here for `abstract String getSeparator(String field);` rather than a `switch` statement in `RequestTemplate`, or I'd drop the enum in favor of a `ParamJoiner` interface (which again has the `String getSeparator(String field)` method).

Reply:
I can use cases for CSV, SSV, PIPES and MULTI.  I am concerned with TAB.  Is the expectation that when that is used a TAB character is present on the query string?  Is that considered safe?

Reply:
Awesome.  Thanks for the info!  Good to know.

Reply:
@rage-shadowman @kdavisk6  I went ahead and refactored this a bit according to your suggestions, and I think it's a bit cleaner now. Instead of the switch statement, I added a `join` method to the enum that contains all the logic for joining together collections of values with the given field name. Now the joining logic in `RequestTemplate` is minimal.

I also renamed MULTI to EXPLODED, which is the term used in the newest edition of OpenAPI. Users will generally not see this value anyway, since that's the default.